python-checks/src/test/resources/checks/argumentType.py

```diff
@@ -243,3 +243,12 @@ def win32_no_support(machine, object, instance, inum, counter):
 def fn_len():
     len(len([1,2,3]))  # Noncompliant
     len(42)  # Noncompliant
+
+def boolean_compatible_with_float() -> None:
```

Comment:
In the same vein, I think we should also have a test with a function that expects an `int` and receives a `bool`, just to make sure this case works as well.